File: datasets/lua-keep-incoder_6b_temp_0.2-1/HumanEval_11_string_xor.lua
-- Input are two strings a and b consisting only of 1s and 0s.
-- Perform binary XOR on these inputs and return result also as a string.
-- >>> string_xor('010', '110')
-- '100'
local function string_xor(a, b)
  local c = ''
  for i = 1, #a do
    c = c .. string.rep(a[i], #b[i])
  end
  return c
end


lu = require('luaunit')

function test_humaneval()
local candidate = string_xor
    lu.assertEquals(candidate('111000', '101010'), '010010')
    lu.assertEquals(candidate('1', '1'), '0')
    lu.assertEquals(candidate('0101', '0000'), '0101')
end

os.exit(lu.LuaUnit.run())

File: datasets/ts-keep-code_davinci_001_temp_0.2/HumanEval_11_string_xor.ts
//Input are two strings a and b consisting only of 1s and 0s.
// Perform binary XOR on these inputs and return result also as a string.
// >>> string_xor('010', '110')
// '100'
function string_xor(a: string, b: string): string {
  const aArr = a.split('').map(Number);
  const bArr = b.split('').map(Number);
  const result = aArr.map((bit, i) => bit ^ bArr[i]);
  return result.join('');
}


declare var require: any;
const assert = require('node:assert');


function test() {
  let candidate = string_xor;
  assert.deepEqual(candidate("111000", "101010"),"010010");
  assert.deepEqual(candidate("1", "1"),"0");
  assert.deepEqual(candidate("0101", "0000"),"0101");
}

test();